quantulum3/disambiguate.py

```diff
@@ -3,41 +3,95 @@
 :mod:`Quantulum` disambiguation functions.
 """
 
+import logging
+
 from . import classifier as clf
 from . import load
 from . import no_classifier as no_clf
 
+_LOGGER = logging.getLogger(__name__)
 
 ###############################################################################
+
+
 def disambiguate_unit(unit_surface, text, lang="en_US"):
     """
     Resolve ambiguity between units with same names, symbols or abbreviations.
     :returns (str) unit name of the resolved unit
     """
-    if clf.USE_CLF:
-        base = clf.disambiguate_unit(unit_surface, text, lang).name
-    else:
-        base = (
-            load.units(lang).symbols[unit_surface]
-            or load.units(lang).surfaces[unit_surface]
-            or load.units(lang).surfaces_lower[unit_surface.lower()]
-            or load.units(lang).symbols_lower[unit_surface.lower()]
-        )
+    units = attempt_disambiguate_unit(unit_surface, text, lang)
+    if units and len(units) == 1:
+        return next(iter(units)).name
+
+    # Change the capitalization of the last letter to find a better match.
+    # Capitalization is sometimes cause of confusion, but the
+    # capitalization of the prefix is too important to alter.
 
-        if len(base) > 1:
-            base = no_clf.disambiguate_no_classifier(base, text, lang)
-        elif len(base) == 1:
-            base = next(iter(base))
+    # If the unit is longer than two prefixes, we set everything to lower
```

Comment:
The whole following flow is convoluted and not concise. I don't really see how there is not a ton of issues here.
- If a unit has no prefix but starts with "m" (like meter) this messes up stuff
- Why do we selectively lower/swapcase certain parts of the surface? Should we not try out all possible capitalizations?
- Why set everything to lower for units longer than two letters (nothing known about prefixes)?

In general, this whole flow needs more explanation like "units that have more than two letters are most likely case insensitive. Only the capitalization of the prefix matters. We therefore only try out lowering the last letters"